### ISSUE TITLE
imgadm incorrectly shows deleted vmadm image

### ISSUE THREAD

**Issue by RSully** [joyent/smartos-live#437]
I had created a zone that ended up failed, and deleted it from `vmadm`. For some reason the image (zones/$uuid) did not get deleted by `vmadm`, and now `imgadm list` shows it as an unknown image.

```
# imgadm list
UUID                                  NAME                    VERSION     OS       PUBLISHED
f067fe9a-4702-46cf-a73a-2daf38b4b20b  -                       -           -        -
```

Looking through `/var/log/vm` it appears that zone creation had originally timed out.

**Comment by RSully** [comment]
I forgot to add, when I attempt to delete it I get this:

```
# imgadm delete f067fe9a-4702-46cf-a73a-2daf38b4b20b 
imgadm delete: error (InternalError): error deleting image "f067fe9a-4702-46cf-a73a-2daf38b4b20b": Error: Command failed: cannot unmount '/zones/f067fe9a-4702-46cf-a73a-2daf38b4b20b': Device busy
cannot unmount '/zones/f067fe9a-4702-46cf-a73a-2daf38b4b20b': Device busy
```

And a listing:

```
# ls -la /zones/f067fe9a-4702-46cf-a73a-2daf38b4b20b/
total 34
drwx------   2 root     root           2 Apr  3 23:09 .
drwxr-xr-x  15 root     root          18 Apr  5 00:36 ..
```

**Comment by bahamat** [comment]
Duplicate of #316.

**Comment by RSully** [comment]
Thanks for the link. @rmustacc will probably be interested in that too then.

**Comment by RSully** [comment]
Is the issue of `vmadm` failing to delete the dataset separate from the issue of `imgadm` displaying incorrect datasets? 

Since this started I have not been able to delete that image (haven't been able to reboot).

**Comment by bahamat** [comment]
Possibly, @joshwilsdon would need to comment.

**Comment by RSully** [comment]
I upgraded to 20150403 image and the zone has disappeared from `imgadm`. However, I don't know if the reboot caused this, or if the fix in 7862381f552cc36f8ebc0b62d9044ca039ab25dd caused this.

**Comment by RSully** [comment]
Rebooted to 20150320 image and dataset shows up in `imgadm`. So this is fixed.

However, #316, the issue of `vmadm` failing to delete/cleanup datasets likely still exists.
